src/main/java/com/edmunds/tools/databricks/maven/UpsertClusterMojo.java

```diff
@@ -53,7 +53,7 @@
 /**
  * Cluster mojo, to perform databricks cluster upsert (create or update through recreation).
  */
-@Mojo(name = "upsert-cluster")
+@Mojo(name = "upsert-cluster", requiresProject = false)
 public class UpsertClusterMojo extends BaseDatabricksMojo {
```

Comment:
I could be remembering wrong, but the problem with doing it this way is that it won't work with sub modules when you have requiresProject = false. This is why there is unfortunately the complexity of a NoProject Mojo for each of the ones that we are electing to run outside of the pom. You should do the same for this one too.

Reply:
Yes, you're right!
I've introduced a new Mojo, refectored unit tests and checked multi-module project scenario along with no-project one.

Reply:
excellent!